src/GraphRequest.ts

```diff
@@ -501,19 +494,29 @@ export class GraphRequest {
 	 * @returns The same GraphRequest instance that is being called with
 	 */
 	public query(queryDictionaryOrString: string | KeyValuePairObjectStringNumber): GraphRequest {
-		const otherURLQueryParams = this.urlComponents.otherURLQueryParams;
+		let paramKey: string;
+		let paramValue: string | number;
 		if (typeof queryDictionaryOrString === "string") {
-			const querySplit = queryDictionaryOrString.split("=");
-			const queryKey = querySplit[0];
-			const queryValue = querySplit[1];
-			otherURLQueryParams[queryKey] = queryValue;
+			/* The query key-value pair must be split on the first equals sign to avoid errors in parsing nested query parameters.
+			 Example-> "/me?$expand=home($select=city)" */
+			const indexOfFirstEquals = queryDictionaryOrString.indexOf("=");
+			paramKey = queryDictionaryOrString.substring(0, indexOfFirstEquals);
+			paramValue = queryDictionaryOrString.substring(indexOfFirstEquals + 1, queryDictionaryOrString.length);
 		} else {
 			for (const key in queryDictionaryOrString) {
 				if (queryDictionaryOrString.hasOwnProperty(key)) {
-					otherURLQueryParams[key] = queryDictionaryOrString[key];
+					paramKey = key;
+					paramValue = queryDictionaryOrString[key];
 				}
 			}
 		}
+		if (oDataQueryNames.indexOf(paramKey) !== -1) {
+			const currentValue = this.urlComponents.oDataQueryParams[paramKey];
+			const isValueAppendable = (paramKey === "$expand" || paramKey === "$select" || paramKey === "$orderby") && currentValue;
```

Comment:
small perf: if `currentValue` comes first, this can be short circuited for cases where `currentValue` is false.

```suggestion
			const isValueAppendable = currentValue && (paramKey === "$expand" || paramKey === "$select" || paramKey === "$orderby");
```